synapse/app/homeserver.py

```diff
@@ -252,23 +253,34 @@ def start_listening(self) -> None:
                     self._listener_http(self.config, listener)
                 )
             elif listener.type == "manhole":
-                _base.listen_manhole(
-                    listener.bind_addresses,
-                    listener.port,
-                    manhole_settings=self.config.server.manhole_settings,
-                    manhole_globals={"hs": self},
-                )
+                if isinstance(listener, TCPListenerConfig):
+                    _base.listen_manhole(
+                        listener.bind_addresses,
+                        listener.port,
+                        manhole_settings=self.config.server.manhole_settings,
+                        manhole_globals={"hs": self},
+                    )
+                else:
+                    raise ConfigError(
+                        "Can not using a unix socket for manhole at this time."
```

Comment:
using → use

(can one not suggest improvements in a commit-by-commit review? :/)

Reply:
(I suppose not? 🤷‍♂️ )
Fixed in [8876780](https://github.com/matrix-org/synapse/pull/15353/commits/88767800a419ff6fb60b4b7489802952ee1a58eb)